rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -373,37 +374,38 @@ class NodeHandle_ : public NodeHandleBase_
   {
     std_msgs::Time t;
     publish(TopicInfo::ID_TIME, &t);
-    rt_time = hardware_.time();
+    rt_time_micros = hardware_.time_micros();
+    rt_time = rt_time_micros / 1000;
   }
 
   void syncTime(uint8_t * data)
   {
     std_msgs::Time t;
-    uint32_t offset = hardware_.time() - rt_time;
+    uint64_t offset = hardware_.time_micros() - rt_time_micros;
 
     t.deserialize(data);
-    t.data.sec += offset / 1000;
-    t.data.nsec += (offset % 1000) * 1000000UL;
+    t.data.sec += offset / 1000000;
+    t.data.nsec += (offset % 1000000) * 1000000UL;
 
     this->setNow(t.data);
     last_sync_receive_time = hardware_.time();
   }
 
   Time now()
   {
-    uint32_t ms = hardware_.time();
+    uint64_t mus = hardware_.time_micros();
     Time current_time;
-    current_time.sec = ms / 1000 + sec_offset;
-    current_time.nsec = (ms % 1000) * 1000000UL + nsec_offset;
+    current_time.sec = mus / 1000000 + sec_offset;
+    current_time.nsec = (mus % 1000000) * 1000000UL + nsec_offset;
```

Comment:
maybe my brain is already in friday evening mode... but shouldn't this be 
```
 current_time.nsec = (mus % 1000000) * 1000UL + nsec_offset;
```

Reply:
You might be right! Thanks for this catch. Unfortunately, I can not test it right away as I don't have my cameras atm...